### ISSUE TITLE
please add a windowed option

### ISSUE THREAD

**Issue by LingKuja** [CookiePLMonster/SilentPatchTOCA2#2]
Like the extremely good SilentPatch for Colin McRae Rally 2, I'd love for an option to use windowed mode rather than fullscreen only.
Alt+Enter does nothing on my end, nor do dxwnd or dgVoodoo help with the situation (especially dgVoodoo, justs screws the coloring rendering everything white and black)

**Comment by CookiePLMonster** [comment]
Are you sure dgVoodoo is a no-go? I personally use it with toca2 just fine (version 2.47.1), incl. windowed mode.

**Comment by LingKuja** [comment]
I'm sure I'm probably not good at using it, although I did dropped all the required DLLs from dgVoodoo\MS to the Toca2\Game directory.
These are my settings and how the game looks (the same error happened on Colin McRae Rally 1 as well), with dgVoodoo
![TC2_2022-02-10_22-36-31](https://user-images.githubusercontent.com/35222780/153501043-7bf113e6-4e6a-4a37-a2ea-bd87e6fb2997.png)
![dgVoodooCpl_2022-02-10_22-35-00](https://user-images.githubusercontent.com/35222780/153501048-f01a0585-1e07-47de-8cf1-ae98c8fc31e0.png) ![dgVoodooCpl_2022-02-10_22-35-09](https://user-images.githubusercontent.com/35222780/153501050-f0401525-0e27-413d-8e9c-afbdc2205ae0.png) ![dgVoodooCpl_2022-02-10_22-35-22](https://user-images.githubusercontent.com/35222780/153501051-788f5867-b125-401e-b411-2a38ff24b630.png)
EDIT: after reading dgVoodoo's docs a bit more I realized my mistake, and now the game is in windowed, but the white/black glich is still present.

While without the dgVoodoo the game is in fullscreen only, 640x480 and with weird black artifacts all over the screen, which aren't static, and don't show in the races.
![TC2_2022-02-10_22-39-54](https://user-images.githubusercontent.com/35222780/153501420-565376cb-199a-4249-b92e-96181cf6bf70.png)

**Comment by CookiePLMonster** [comment]
This is my settings (Glide section doesn't matter for TOCA), works perfectly fine.
![image](https://user-images.githubusercontent.com/7947461/153510380-e2b59c6a-c343-48b4-bbdf-5be4d85b617a.png)
![image](https://user-images.githubusercontent.com/7947461/153510319-6f1daea3-4993-47e5-ab64-e2208638f343.png)

**Comment by LingKuja** [comment]
probably my pc hates dgvoodoo or something, because even after copying your settings, it still looks black and white :/

**Comment by CookiePLMonster** [comment]
I see that dgVoodoo hosts old builds, maybe you can try out 2.75.1?
http://dege.freeweb.hu/dgVoodoo2/dgVoodoo2/

If that works for you, I'd need to figure out how to submit a regression report.

**Comment by LingKuja** [comment]
nope, I copied the 2.75.1 dlls from MS\x86 to Toca2\Game (replacing the old ones), deleted the dgvoodoo.conf and changed it with the one from 2.75.1 but it still looks white and black. maybe it's the graphics card or the operative system I'm using?
(windows 10 ltsc 2021 IoT)

**Comment by CookiePLMonster** [comment]
It could be the GPU, yes - what GPU do you have?

**Comment by LingKuja** [comment]
It's the Sapphire Pulse AMD Radeon RX 5700 XT

**Comment by CookiePLMonster** [comment]
I'm really not sure in this case, maybe DxWnd or something like this would be of use? Unlike CMR2.0, I don't think TOCA 2 has a native windowed mode left in the code, but I don't remember for sure.

**Comment by LingKuja** [comment]
I've already tested DxWnd extensively before opening the thread, I've used it in the past because of TDU and Outrun 2006, but it doesn't work here :/
Maybe it works on Colin McRae Rally 1.
But then again, it may be just me.
by the way, thank you for your patience :)

**Comment by CookiePLMonster** [comment]
All I could think of is AMD being AMD, maybe you happen to be on a broken driver. No idea to be honest.

**Comment by LingKuja** [comment]
Yes, it probably is just AMD being AMD. Oh well.
